walletkit_client.go

```diff
@@ -737,28 +738,69 @@ func (m *walletKitClient) ListSweepsVerbose(ctx context.Context,
 	return result, nil
 }
 
+// BumpFeeOption customizes a BumpFee call.
+type BumpFeeOption func(*walletrpc.BumpFeeRequest)
+
+// WithImmediate is an option for enabling the immediate mode of BumpFee. The
+// sweeper will sweep this input without waiting for the next batch.
+func WithImmediate() BumpFeeOption {
+	return func(r *walletrpc.BumpFeeRequest) {
+		r.Immediate = true
+	}
+}
+
+// WithTargetConf is an option for setting the target_conf of BumpFee. If set,
+// the underlying fee estimator will use the target_conf to estimate the
+// starting fee rate for the fee function. The value of feeRate passed to
+// BumpFee will be ignored.
```

Comment:
The conf-target only reflects this behaviour for the 18.5 release, before the conf target had a different meaning:

https://github.com/lightningnetwork/lnd/pull/9470

Reply:
I added a check that LND is at least 0.18.5 only if WithTargetConf is used. This commit can be undone after minimalCompatibleVersion is bumbed to 0.18.5.